typescript/phoenix-sdk/src/utils/market.ts

```diff
@@ -861,3 +862,64 @@ export function getMarketExpectedOutAmount({
 
   return expectedUnitsReceived;
 }
+
+/**
+ * Returns the expected amount in for a given desired amount out
+ *
+ * @param ladder
```

Comment:
rename ladder --> uiLadder, and mention that prices are in qu/rbu, quantity is in rbu rather than the market-native units of ticks and base lots